cli/dcnios-cli.py

```diff
@@ -19,175 +19,199 @@
 
-def newProcessInfo(name):
-    print("New Process group: "+ str(name))
 
+def newProcessInfo(name):
+    print("New Process group: " + str(name))
 
 
-parser = argparse.ArgumentParser(prog='ProgramName', description='What the program does', epilog='Text at the bottom of help')
+parser = argparse.ArgumentParser(
+    prog='ProgramName',
+    description='What the program does',
+    epilog='Text at the bottom of help')
 parser.add_argument('option')
-parser.add_argument('-f')  
+parser.add_argument('-f')
 parser.add_argument('--host')
-parser.add_argument('--user',"-u")
-parser.add_argument('--password',"-p")
-parser.add_argument('--processGroup',"-pg")
-parser.add_argument('--component',"-c")
-parser.add_argument('--seconds',"-s")
+parser.add_argument('--user', "-u")
+parser.add_argument('--password', "-p")
+parser.add_argument('--processGroup', "-pg")
+parser.add_argument('--component', "-c")
+parser.add_argument('--seconds', "-s")
 args = parser.parse_args()
-if args.option  is None and args.f is None:
+if args.option is None and args.f is None:
     print("File parameter not found")
     exit(1)
 # Open the file and load the file
-elif args.option is not None  and args.f is not None:
+elif args.option is not None and args.f is not None:
     with open(args.f) as f:
         data = yaml.load(f, Loader=SafeLoader)
-        nifi_endpoint=data["nifi"]["endpoint"]
-        nifi_user=data["nifi"]["user"]
-        nifi_password=data["nifi"]["password"]
-        nifi=Nifi(nifi_endpoint,nifi_user,nifi_password)
-        if args.option == "apply": 
+        nifi_endpoint = data["nifi"]["endpoint"]
+        nifi_user = data["nifi"]["user"]
+        nifi_password = data["nifi"]["password"]
+        nifi = Nifi(nifi_endpoint, nifi_user, nifi_password)
+        if args.option == "apply":
             if "dCache" in data["nifi"]:
                 for dcache in data["nifi"]["dCache"]:
-                    dcachecontent=prepareforAll(dcachefile)
-                    nifi.create(dcache["name"], dcachecontent )
-                    command="simple-client.py --state /state/"+dcache["statefile"]+" --endpoint "+ \
-                        dcache["endpoint"]+" --user "+dcache["user"]+" --password "+ \
-                        dcache["password"]+" "+ dcache["folder"]
-                    nifi.changeVariable(dcache["name"],"command",command)
+                    dcachecontent = prepareforAll(dcachefile)
+                    nifi.create(dcache["name"], dcachecontent)
+                    command = "simple-client.py --state /state/" \
+                        + dcache["statefile"] + " --endpoint " \
+                        + dcache["endpoint"] \
+                        + " --user "+dcache["user"] + " --password " \
+                        + dcache["password"] + " " + dcache["folder"]
+                    nifi.changeVariable(dcache["name"], "command", command)
                     newProcessInfo(dcache["name"])
                     updateComponent(dcache)
```

Comment:
I would create an external function for this block of code and the same for the rest of the values (dCache, OSCAR, generic...). The resulting code would be easier to read and maintain.